utils/gcp/kms/main.tf

```diff
@@ -0,0 +1,73 @@
+
+data "google_client_config" "current" {}
+
+###### SECTION - IAM POLICY KMS
+
+resource "google_kms_crypto_key_iam_member" "crypto_key_role" {
+  for_each      = var.crypto_key_roles != null ? merge([for role, members in var.crypto_key_roles : { for member in members : "${role}-${member}" => { role = role, member = member } }]...) : {}
+  crypto_key_id = google_kms_crypto_key.kms_crypto_key.id
+  role          = each.value.role
+  member        = each.value.member
+}
+
+###### SECTION - IAM POLICY KMS RING
+
+resource "google_kms_key_ring_iam_member" "key_ring_role" {
+  for_each    = var.key_ring_roles != null ? merge([for role, members in var.key_ring_roles : { for member in members : "${role}-${member}" => { role = role, member = member } }]...) : {}
+  key_ring_id = google_kms_key_ring.key_ring.id
+  role        = each.value.role
+  member      = each.value.member
+}
+
+###### SECTION - KMS KEY RING
+
+resource "google_kms_key_ring" "key_ring" {
+  name     = var.kms_key_ring_name
+  location = data.google_client_config.current.region
```

Comment:
Location of a key ring is not necessarily a region, it can be multiple regions, a continent, or global.